recipe/meta.yaml

```diff
@@ -1,17 +1,16 @@
 {% set name = "helics" %}
-{% set version = "2.0.0" %}
-{% set checksum = "aa1a7be9032d1e7ad886bced08b4e0e8d3ffaf13fa21d1239e91c47e8a9dd677" %}
+{% set version = "2.1.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://github.com/GMLC-TDC/HELICS-src/archive/v{{ version }}.tar.gz
-  sha256: {{ checksum }}
+  git_url: https://github.com/GMLC-TDC/HELICS.git
+  git_rev: v{{ version }}
```

Comment:
See https://conda-forge.org/docs/maintainer/adding_pkgs.html#tarballs-no-repos

Reply:
Git submodules are used -- GitHub tarballs don't have the required metadata to check them out.

Reply:
Which submodules?

Reply:
Asio - https://github.com/chriskohlhoff/asio
JsonCpp fork with several fixes - https://github.com/phlptp/jsoncpp.git
fmtlib - https://github.com/fmtlib/fmt.git
gmlc::containers - https://github.com/GMLC-TDC/containers.git
gmlc::concurrency - https://github.com/GMLC-TDC/concurrency.git
gmlc::utilities - https://github.com/GMLC-TDC/utilities.git

Reply:
It might be possible to initialize a git repo in the folder from the extracted tarball, then run the git command needed to clone those repositories using the .gitmodules file, but that would get the most recent commit rather than the version that a tagged release was made with.

Reply:
If you are using code from these repos, you need to add the licenses. `about: license_file` can be a yaml list to include multiple licenses.
What's the issue with jsoncpp? Have you sent these fixes sent upstream?

Reply:
There were fixes for warnings, and the addition of an exported CMake OBJECT library for JsonCpp (to avoid needing to install a separate static library for JsonCpp that might conflict with an already installed version; relying on a system install wasn't an option because some targeted distros only have packages for very old version of JsonCpp). Eventually I think the plan is to send the changes upstream, if they will take them.

I was looking for an example of a feedstock [searched for go packages, since vendoring 3rd party dependencies is fairly common] that uses multiple licenses file entries (which I haven't found yet) to see how they handle pointing to 3rd party library license files or if they just list the home page url, license, and summary. I came across the [glide-feedstock](https://github.com/conda-forge/go-glide-feedstock/blob/master/recipe/meta.yaml) and [blosc-feedstock](https://github.com/conda-forge/blosc-feedstock/blob/master/recipe/meta.yaml) which use vendoring to distribute their 3rd party dependencies in a [vendor folder](https://github.com/Masterminds/glide/tree/master/vendor/github.com) or [internal libs folder](https://github.com/Blosc/c-blosc/tree/master/internal-complibs/snappy-1.1.1). It seems tensorflow also uses quite a few third party libraries.

I guess there are two things I'm wondering about now from looking for examples:
- For listing 3rd party libraries in a conda package, is the url, license type, and summary sufficient? Or do we need to also figure out where all of the license files live in their respective repositories (which could change if they move things around in the upstream repository)?
- Is there a difference between the way that we are doing things with git submodules that makes so listing out all of the licenses of 3rd party libraries is required compared to what these other packages are doing?  Or is it more a matter of these other packages not following the best practice of listing every 3rd party library license either, and an effort is being made to gradually get all packages in compliance?

Reply:
> There were fixes for warnings, and the addition of an exported CMake OBJECT library for JsonCpp (to avoid needing to install a separate static library for JsonCpp that might conflict with an already installed version; relying on a system install wasn't an option because some targeted distros only have packages for very old version of JsonCpp)

We have version 1.8.4. Is that enough? Is there a way to optionally use a system install?

> I was looking for an example of a feedstock [searched for go packages, since vendoring 3rd party dependencies is fairly common] that uses multiple licenses file entries (which I haven't found yet)

This is a new feature. https://github.com/conda/conda-build/pull/3600/files#diff-3cd4e1d5ae8274c9e3f742ab550d60d6R12 . We used to concatanate them https://github.com/conda-forge/pyfmmlib-feedstock/blob/master/recipe/build.sh#L10-L16

> Or do we need to also figure out where all of the license files live in their respective repositories (which could change if they move things around in the upstream repository)?

Unfortunately you need the license file to comply with most open source licenses. For example asio is Boost licensed and the copyright and the license text has to be packaged.

>  Or is it more a matter of these other packages not following the best practice of listing every 3rd party library license either, and an effort is being made to gradually get all packages in compliance?

It's this one. I couldn't have said it better myself.

Reply:
Version 1.8.4 of JsonCpp would be enough, but right now there isn't a way to use a system install (some Json changes are on the TODO list).

Thanks for the new feature link.

I guess changes for license compliance with cpack installers/packages will be a new todo for the next release. Interesting that the BSD license (which most of our libraries use) is more restrictive with its binary redistribution clause than the Boost license which has an exception for machine-executable object code. Boost/BSD/MIT are more different than I thought.